### ISSUE TITLE
Ponder including `Cargo.lock` in version control and releases

### ISSUE THREAD

**Issue by rgacogne** [cloudflare/quiche#1791]
Hi!

I was wondering if you ever considered including a `Cargo.lock` file in your repository and releases? There are pros and cons to doing that, as described in [1], but it helps a lot in getting reproducible builds. It would allow me to move forward with making the Quiche package in Arch Linux [2] reproducible, for example.

[1]: https://doc.rust-lang.org/nightly/cargo/faq.html#why-have-cargolock-in-version-control
[2]: https://archlinux.org/packages/extra/x86_64/quiche/

**Comment by birneee** [comment]
I agree that having reproducible builds of the libraries and tooling in this repository would be very useful. 
I encountered a similar issue with Nix builds, which requires specific dependencies. A Cargo.lock file would simplify packaging quiche with nix and possibly add it to nixpkgs.
You can see my fork here: https://github.com/birneee/quiche/tree/nix

**Comment by Squawkykaka** [comment]
Yes i am also packaging it for nix and this would help a lot
